resources/assets/tconstruct/blockstates/rack.json

```diff
@@ -0,0 +1,22 @@
+{
+    "forge_marker": 1,
+    "defaults": {
+        "model": "tconstruct:item_rack"
+    },
+    "variants": {
+        "facing": {
+            "down_z": { "transform": { "translation": [ 0, -0.75, 0 ] } },
+            "down_x": { "transform": { "rotation": { "y": 270 }, "translation": [ 0, -0.75, 0 ] } },
+            "up_z":   {},
+            "up_x":   { "y": 90 },
+            "north":  { "transform": { "translation": [ 0, 0, -0.375 ] } },
+            "south":  { "transform": { "translation": [ 0, 0,  0.375 ] } },
+            "east":   { "transform": { "rotation": { "y": 270 }, "translation": [  0.375, 0, 0 ] } },
+            "west":   { "transform": { "rotation": { "y": 270 }, "translation": [ -0.375, 0, 0 ] } }
+        },
+        "drying": {
+            "true": { "model": "tconstruct:drying_rack" },
+            "false": {}
+        }
+    }
```

Comment:
Add [inventory variant](https://github.com/SlimeKnights/TinkersConstruct/blob/master/resources/assets/tconstruct/blockstates/faucet.json#L14), can be defined like any other blockstate too